File: server/paper/src/main/java/su/plo/voice/paper/connection/PaperServerServiceChannelHandler.java
package su.plo.voice.paper.connection;

import org.bukkit.entity.Player;
import org.bukkit.plugin.messaging.PluginMessageListener;
import org.jetbrains.annotations.NotNull;
import su.plo.voice.api.server.player.VoiceServerPlayer;
import su.plo.voice.server.BaseVoiceServer;
import su.plo.voice.server.connection.BaseServerServiceChannelHandler;

import java.io.IOException;

public final class PaperServerServiceChannelHandler extends BaseServerServiceChannelHandler implements PluginMessageListener {

    public PaperServerServiceChannelHandler(@NotNull BaseVoiceServer voiceServer) {
        super(voiceServer);
    }

    @Override
    public void onPluginMessageReceived(@NotNull String channelName, @NotNull Player player, @NotNull byte[] message) {
        try {
            VoiceServerPlayer voicePlayer = voiceServer.getPlayerManager().wrap(player);

            handlePacket(voicePlayer, message);
        } catch (IOException e) {
            e.printStackTrace();
        }
    }
}

File: server/paper/src/main/kotlin/su/plo/voice/paper/connection/PaperServerServiceChannelHandler.kt
package su.plo.voice.paper.connection

import org.bukkit.entity.Player
import org.bukkit.plugin.messaging.PluginMessageListener
import su.plo.voice.server.BaseVoiceServer
import su.plo.voice.server.connection.BaseServerServiceChannelHandler
import java.io.IOException

class PaperServerServiceChannelHandler(voiceServer: BaseVoiceServer) : BaseServerServiceChannelHandler(voiceServer),
    PluginMessageListener {

    override fun onPluginMessageReceived(channelName: String, player: Player, message: ByteArray) {
        try {
            val voicePlayer = voiceServer.playerManager.wrap(player)
            handlePacket(voicePlayer, message)
        } catch (e: IOException) {
            e.printStackTrace()
        }
    }
}
